ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -963,6 +1001,176 @@ def setup_jmt(self, curpt, target_velocity, accel_ratio=1.0,
 
         return jmt_ptr-1
 
+    def get_max_accel(self, ptr_id):
+        # Use JMT to figure out shortest speedup distance with
+        # maximum jerk of < 5.0 at 0.1 s after change in
+        # direction of acceleration
+        curpt = self.waypoints[ptr_id]
+        if curpt.get_v() >= self.default_velocity * 0.975:
+            return (0.5, 0.5)
+
+        timer_start = rospy.get_time()
+        too_short = False
+
+        if curpt.get_a() < 0.0:  # might prefer to check previous state to see if was in slowdown
+            rospy.logwarn("look for max accel since currently decelerating at {:3.2f}".format(curpt.get_a()))
+            accel_rate = 0.6
+            max_jerk = self.max_jerk * 0.90
```

Comment:
Since distance to speed up doesn't need to be as short as possible, should the jerk criteria be `self.max_desired_jerk` instead to be a little smoother?